### ISSUE TITLE
check boxes status cannot be changed (internittendly) 

### ISSUE THREAD

**Issue by ajay-gh** [laurent22/joplin#4490]
## Environment

Joplin version: 1.7.10
Platform: Mac OS
OS specifics: Mojave up-to-date

## Steps to reproduce

1. It's intermittent, meaning one cannot always reproduce it, but ... 
2. select a note with check-boxes at the beginning of the line, select the WYSIWYG editor
3. try to click ("check" the check box)

## Describe what you expected to happen
the check mark should toggle from "done" to "not yet done"

## Logfile

[log.txt](https://github.com/laurent22/joplin/files/5938723/log.txt)

Followed debugging instruction, and I could reproduce the error - luckily. While clicking (without success) on check-boxes the following text appeared inside the console:
11:39:23: Loaded note: {id: "3f2d3b317dea41b5bf7898b3a21c78b2", parent_id: "157e9b2a9fe54dd3a1a1246aa6664e46", title: "next AIs", body: "- [x] continue XXX↵    ↵- [x] \## // post f…eck XXX is still worth it↵    ↵- [x] pay XXX", created_time: 1605174847376, …}

Note: XXX in the last paragraph denotes personal information (text) which I replaced.

The only other unusual output in the console window is this:
DevTools failed to load SourceMap: Could not load content for file:///Applications/Joplin.app/Contents/Resources/app.asar/node_modules/tinymce/skins/ui/oxide/skin.min.css.map: 
DevTools failed to load SourceMap: Could not load content for file:///Applications/Joplin.app/Contents/Resources/app.asar/node_modules/tinymce/skins/ui/oxide/content.min.css.map: 
DevTools failed to load SourceMap: Could not load content for file:///Applications/Joplin.app/Contents/Resources/app.asar/node_modules/tinymce/skins/content/default/content.min.css.map: 
DevTools failed to load SourceMap: Could not load content for file:///Applications/Joplin.app/Contents/Resources/app.asar/pluginAssets/mermaid/mermaid.min.js.map:

**Comment by laurent22** [comment]
I use checkboxes all the time in the Rich Text editor and never had this problem. Could you provide an example note with which you got this issue please? (exporting it as JEX would be best to preserve the exact structure).

**Comment by ajay-gh** [comment]
voilà 
[test.jex.zip](https://github.com/laurent22/joplin/files/5939481/test.jex.zip)

**Comment by laurent22** [comment]
Unfortunately I cannot replicate the bug with this file.

**Comment by ajay-gh** [comment]
did you see the short movie which I posted on discourse ?
what else can I do (other than sending you my complete profile folder, ... which I can't do) ?

**Comment by coderrsid** [comment]
can you please mention the link to discourse in which you posted the video to reproduce a bug?

**Comment by ajay-gh** [comment]
https://discourse.joplinapp.org/t/check-box/14628?u=ajay

**Comment by ajay-gh** [comment]
One extra insight. Today I found another frozen checkbox in my Notes. Went to the markdown editor, and replaced the corresponding [x] with a [ ] (or vice versa). Back to the wys editor the status of the checkbox has changed too, but still no change can be orchestrated within the wys editor. My feeling is that this does only happen after a certain number of edits (and may be syncs) within the wys editor - which is my typical ops mode, and would explain why I have the problem, and others don't.

**Comment by laurent22** [comment]
Yes could be. Although I use checkboxes quite extensively in the rich text editor and never had this issue so far. Do you use some custom CSS or plugins that could affect rendering?

**Comment by ajay-gh** [comment]
No CSS, only one plugin called Simple Backup. But the problem occurred before I installed that plugin. 
Of course, with the work-around discovered yesterday (see above) it is less a problem than before. Regards

**Comment by coderrsid** [comment]
I am sorry but not able to reproduce this.

**Comment by ron-debajyoti** [comment]
This is weird because I see this bug in v1.8.0 Joplin dev in Ubuntu 18.04.  
For me, I am unable to check/uncheck any box in Rich Text Editor.

**Comment by ron-debajyoti** [comment]
Update: I got a weird bug in my case. I can't check/uncheck the box when the `Checkbox-List` option in the Toolbar isn't visible and under `More`.
When I expand the editor so that all the options are visible, the list starts working.
The [joplin](https://discourse.joplinapp.org/t/check-box/14628?u=ajay) forum for this is closed for inactivity, so perhaps I could open a new topic or start working on this and make a PR immediately.

**Comment by ron-debajyoti** [comment]
> Update: I got a weird bug in my case. I can't check/uncheck the box when the `Checkbox-List` option in the Toolbar isn't visible and under `More`.
> When I expand the editor so that all the options are visible, the list starts working.

A video demonstrating it would perhaps be helpful.

https://user-images.githubusercontent.com/22571664/113483723-8ae8a200-94c2-11eb-93f8-6a4b42581162.mp4

**Comment by ajay-gh** [comment]
I can confirm the same "work around" on Macos. When I expand the editor options (toolbar) the problem is gone ... as long as the toolbar remains unfolded. It is however not just a linked to this. Toolbar folded or not, those notes where the checkboxes work as expected do so all the time.

**Comment by ajay-gh** [comment]
A few pre-releases down the line (we are now at v 1.8.5), I think this is worth some attention AND time to get rid of the "cannot-reproduce" label. This has now been confirmed on several devices and systems. What can we do to show it to you, Laurent ?

**Comment by tomasz1986** [comment]
I have the same problem in Windows, so this is not Mac-specific.

It is very easy to reproduce. Just shrink the Joplin window a little bit in width, so that the checkbox list option becomes hidden under the three dots menu. As soon as this is the case, it is no longer possible to interact with any of them, unless you click on the three dots and display the hidden options.

I have only encountered the problem, because I have two monitors, with the second one being an older 1280x1024 screen. On that one, I cannot use the checkboxes. On the other one, which is 1920x1200, everything works normally.

**Comment by markkuit** [comment]
I'm experiencing the same issue on Joplin v2.5.12 (linux), where I cannot check/uncheck boxes if the option is hidden in the triple-dot menu, but I can if I expand it.

**Comment by stevemagruder** [comment]
I just realized I'm having this same issue on Joplin 2.8.8 (Windows 10) while I didn't see it on my other Windows 11 system. My screen real estate is scrunched on the Win10 system so that I see the ellipses in the toolbar.
Before I click the ellipses, I am unable to toggle a checkbox in the note, but after I click the ellipses, I then have the ability to toggle them.
